pkg/segment/query/pqs/meta/pqsmeta.go

```diff
@@ -160,17 +156,22 @@ func removePqmrFilesAndDirectory(pqid string) error {
 	return nil
 }
 
-func DeleteSegmentFromPqid(pqid string, segKey string) {
+// This function will remove the PQMRFiles and directory if there are no segments left in the PQID
+func BulkDeleteSegKeysFromPqid(pqid string, segKeyMap map[string]bool) {
 	pqFname := getPqmetaFilename(pqid)
 	emptyPQS, err := getAllEmptyPQSToMap(pqFname)
 	if err != nil {
-		log.Errorf("DeleteSegmentFromPqid: Failed to get empty PQS data from file at %s: Error=%v", pqFname, err)
+		log.Errorf("BulkDeleteSegKeysFromPqid: Failed to get empty PQS data from file at %s: Error=%v", pqFname, err)
 	}
-	delete(emptyPQS, segKey)
+
+	for segKey := range segKeyMap {
+		delete(emptyPQS, segKey)
+	}
+
```

Comment:
if the `len of emptyPQS` is not zero then don't we have to write the .pqmeta file with the leftover entries ?

Reply:
We are writing there is a call for `writeEmptyPqsMapToFile` function at Line: 178

Reply:
I see